### PR TITLE
feat: add ranged enemy animations

### PR DESCRIPTION
**Pull request by JoenvandeVorle**
## Description

Add idle, walking, loading, and shooting animations to the ranged enemy.

Also fixes-a two bugs: The melee weapon hitbox no moves around based on the melee animation, and the double damage bug for ranged enemies is also fixed.

Also moved some navmesh folders into the archived folder

## Testability

Go to `AddRangedAnimations` in the archived scenes folder. In there you will find a respawning ranged enemy that does 0 damage, so you can see all the animations there.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

**Review by SSelinn — APPROVED** [review]
LGTM

**Review by SSelinn — APPROVED** [review]
yeye

File: aplib.net-demo/Assets/Scenes/AddRangedAnimations.unity
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scenes/AddRangedAnimations.unity:1]: Not sure what this scene is for, but when running it I only get a menu that does not go away.

File: aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs [hunk 1]
```diff
@@ -18,6 +23,10 @@ public class RangedEnemy : RespawningEnemy
 
     private RangedWeapon _rangedWeapon;
 
+    private Animator _animator;
+
+    private Rigidbody _rigidbody;
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs:28]: This variable appears to never be used.

File: aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs [hunk 2]
```diff
@@ -27,8 +36,10 @@ protected override void Start()
         _rangedWeapon = GetComponentInChildren<RangedWeapon>();
         _rangedWeapon.Initialize(_damagePoints, _targetTag, transform, _attackRange);
         _attackTimer = gameObject.AddComponent<Timer>();
-        _attackTimer.SetExactTime(_attackCooldown);
+        _attackTimer.SetExactTime(_attackCooldown + _attackChargeUp);
         _pathFind.SetStoppingDistance(_attackRange - 1f);
+        _animator = GetComponent<Animator>();
+        _rigidbody = GetComponent<Rigidbody>();
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs:42]: This line can be removed if the rigidbody var is.

File: aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs [hunk 3]
```diff
@@ -42,15 +53,30 @@ protected override void Update()
         Debug.DrawRay(transform.position, Vector3.up * 20, Color.green);
 
         // If the target is not directly visible but within vision range, move closer to the target.
-        if (!_rangedWeapon.EnemiesInLineOfSight() && _pathFind.GoalWithinRange(_visionRange)) _pathFind.SetStoppingDistance(1f);
+        if (!_rangedWeapon.EnemiesInLineOfSight() && _pathFind.GoalWithinRange(_visionRange))
+        {
+            _pathFind.SetStoppingDistance(1f);
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs:58]: Should that 1f value be defined above so it can be set better as well?

**Review by QuakeEye — CHANGES_REQUESTED** [review]

File: aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs (hunk 3 above)
Comment (JoenvandeVorle) [reply aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs:58]: no, this is just an arbitrary small number to make the enemy move closer

File: aplib.net-demo/Assets/Scenes/AddRangedAnimations.unity
Comment (JoenvandeVorle) [reply aplib.net-demo/Assets/Scenes/AddRangedAnimations.unity:1]: Forgot to set the scene name in the canvasmanager, should work now

File: aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs (hunk 3 above)
Comment (QuakeEye) [reply aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs:55]: But it is used in multiple places, should that not be defined?

**Review by QuakeEye — CHANGES_REQUESTED** [review]
Besides the previous comment all is well

File: aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs [hunk 4]
```diff
@@ -87,10 +87,9 @@ public override void UseWeapon()
             // Will damage only enemies and the ray will stop when it hits an object that is not an enemy.
             foreach (RaycastHit hit in _orderedHits)
             {
+                // Stop when the ray hits an object that is not an enemy
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs:90]:
```suggestion
                // Stop when the ray hits an object that is not an enemy.
```

File: aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs [hunk 5]
```diff
@@ -105,6 +104,9 @@ public override void UseWeapon()
         /// <returns>True if the first object hit is an enemy.</returns>
         public bool EnemiesInLineOfSight()
         {
+            // Fire a debug ray in the line of sight
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs:107]:
```suggestion
            // Fire a debug ray in the line of sight.
```

**Review by QuakeEye — CHANGES_REQUESTED** [review]

**Review by QuakeEye — APPROVED** [review]
image

**Pull request merged**